File: src/com/suushiemaniac/cubing/alglib/transform/mirror/MegaminxMirror.java
package com.suushiemaniac.cubing.alglib.transform.mirror;

import com.suushiemaniac.cubing.alglib.move.MegaminxMove;
import com.suushiemaniac.cubing.alglib.move.Move;

public enum MegaminxMirror implements Mirror {
    R_GRIP_AXIS;

    @Override
    public MegaminxMove mirror(Move origin) {
        if (!(origin instanceof MegaminxMove)) return null;
        else return ((MegaminxMove) origin).inverse();
    }

    @Override
    public MegaminxMove transform(Move origin) {
        return this.mirror(origin);
    }
}

File: src/main/kotlin/com/suushiemaniac/cubing/alglib/transform/mirror/MegaminxMirror.kt
package com.suushiemaniac.cubing.alglib.transform.mirror

import com.suushiemaniac.cubing.alglib.move.Move

enum class MegaminxMirror : Mirror {
    R_GRIP_AXIS;

    override fun mirror(origin: Move): Move {
        return origin.inverse()
    }
}